src/tools/fuzzing.h

```diff
@@ -669,6 +683,11 @@ class TranslateToFuzzReader {
         : wasm(wasm), parent(parent) {}
 
       void visitExpression(Expression* curr) {
+        // If this expression can possibly contain a dangling pop, don't
+        // consider this as a candidate for the mutation
+        if (containsChild<Pop>(curr, 2)) {
```

Comment:
Why are these limited to depth 2?

Reply:
Because all cases of `pop` generation in `makeCatch` puts a `pop` within depth 2 compared to the instruction within a `catch` block. For example, when we have something like this code
```wast
(try
  (do ...)
  (catch
    (drop (exnref.pop))
  )
)
```

Here we dropped `exnref.pop` but in `makeCatch` we do four different things with `exnref.pop`. But all of them has `exnref.pop` as their immediate children (which is depth 1. 2 was just a bit of margin or precaution). The purpose here is we shouldn't separate the expression that has `exnref.pop` as its child (e.g. `drop` here) from `catch`. It doesn't matter when we are given an expression of a larger scope, for example the whole `try` here as an argument, because that does not separate `exnref.pop`-owning expression from its corresponding `catch` anymore.

Reply:
FYI I fixed it to 1, because `makeCatch` generates all `pop`s at depth 1 (so if depth 2 ever occurs for some reason, it's likely a bug, in which case depth 2 is not safe anyway). And I think making it depth 2 also can prevent some possible moves.

Reply:
Sorry nevermind I changed it back to 2. Turned out we _need_ 2, because there is an unseen `block` within a catch block which should not be a candidate for replacements.